documentation/design-docs/ipc-protocol.md

```diff
@@ -669,42 +653,427 @@ Payload
+
+> Note available for .NET 10.0 and later.
 
 #### Inputs:
 
-Header: `{ Magic; 28; 0x0201; 0x0000 }`
+Header: `{ Magic; 20 + Payload Size; 0x0206; 0x0000 }`
 
-* `ulong sessionId`: The ID for the streaming session to stop
+#### Streaming Session Payload:
+* `uint output_format`: 0
+* `uint circularBufferMB`: The size of the circular buffer used for buffering event data
+* `uint format`: 0 for the legacy NetPerf format and 1 for the NetTrace V4 format
+* `ulong rundownKeyword`: Indicates the keyword for the rundown provider
+* `bool requestStackwalk`: Indicates whether stacktrace information should be recorded.
+* `array<streaming_provider_config> providers`: The providers to turn on for the session
 
-#### Returns:
+The `streaming_provider_config` is composed of the following data:
+* `ulong keywords`: The keywords to turn on with this provider
+* `uint logLevel`: The level of information to turn on
+* `string provider_name`: The name of the provider
+* `string filter_data`: (Callback filter information) or (length = `0`)
+* `event_filter filter`: Rules for filtering this provider's Event IDs, applied after `keyword`/`logLevel`, using an allow/deny list or (length = `0`).
 
-Header: `{ Magic; 28; 0xFF00; 0x0000 }`
+An `event_filter` is comprised of the following data:
+* `bool allow`: 0 for deny list, 1 for allow list
+* `array<uint> event_ids`: List of Event IDs to deny or allow.
 
-* `ulong sessionId`: the ID for the streaming session that was stopped
+See [event_filter serialization examples](#event_filter)
 
+#### User_events Session Payload:
+* `uint output_format`: 1
+* `ulong rundownKeyword`: Indicates the keyword for the rundown provider
+* `array<user_events_provider_config> providers`: The providers to turn on for the session
 
-##### Details:
+The `user_events_provider_config` is composed of the following data:
+* `ulong keywords`: The keywords to turn on with this provider
+* `uint logLevel`: The level of information to turn on
+* `string provider_name`: The name of the provider
+* `string filter_data`: (Callback filter information) or (length = `0`)
+* `event_filter filter`: Rules for filtering this provider's Event IDs, applied after `keyword`/`logLevel`, using an allow/deny list or (length = `0`).
+* `tracepoint_config config`: Maps Event IDs to tracepoints. If an Event ID is excluded by `event_filter`, it will not be written to any tracepoint.
 
-Inputs:
-```c
-Payload
-{
-   ulong sessionId
-}
+An `event_filter` is comprised of the following data:
+* `bool allow`: 0 for deny list, 1 for allow list
+* `array<uint> event_ids`: List of Event IDs to deny or allow.
+
+See [event_filter serialization examples](#event_filter)
+
+A `tracepoint_config` is comprised of the following data:
```

Comment:
Will we have a default mapping for all events defined in runtime providers, ClrEtwAll.man and this is only needed for custom events?